### ISSUE TITLE
ASP.NET Web core app + installing MVC fails during restore.

### ISSUE THREAD

**Issue by rrelyea** [NuGet/Home#3808]
SCENARIO
1.  New ASP.NET Core Web Application (.NET Core) > Empty
2.  Add Microsoft.AspNetCore.Mvc package using Manage NuGet packages dialog

ISSUES
1.  When I tried earlier, I likely selected the wrong package to install – I typed ‘mvc’ and chose ‘Microsoft.AspNet.Mvc’ by mistake instead of finding ‘Microsoft.AspNetCore.Mvc’. Package install failed and it also rolled back the changes, which is pretty good.
   • Ideally, ‘Microsoft.AspNet.Mvc’ will not even show for ASP.NET Core projects but this can wait until post-Dev15RC. 

> ...
> Package Microsoft.ASpnet.Razor 3.2.3 is not compat with netcoreapp1.0
> ...  
>  Error: one or more packages are incompatible with .NETCOreApp, version=v1.0.
> Package restor failed. Rolling back package changes...
> 
> (rob has image in email)
1.  Packages seems to be always added to the top of package references in csproj. It will be nice if alphabetical order is preserved. I noticed that the template itself does not have the packages ordered alphabetically – I’ll follow-up on those. @Rob, AspNetCore.Mvc and AspNetCore.StaticFiles are the packages that I added using Manage NuGet Packages dialog. Did they get added to the top just because the references from the template were already out of order?

Image that won't paste in shows the following order of  PackRefs. (Rob has email with image)
Mvc, then staticfiles, then netcore.app, then net.sdk.web, then aspnetcore.diagnostics

**Comment by rrelyea** [comment]
Internal email:
Ok, now that I have time to read the bug and answer Ankit’s questions, I see that I misunderstood here. It works, but you only had minor issues.

1)  Ideally, ‘Microsoft.AspNet.Mvc’ will not even show for ASP.NET Core projects but this can wait until post-Dev15RC.
RR: this is a much desired feature that isn’t in scope for Dev15.  It involves client and server work. It isn’t small.

2)  Packages seems to be always added to the top of package references in csproj. It will be nice if alphabetical order is preserved. I noticed that the template itself does not have the packages ordered alphabetically – I’ll follow-up on those. @Rob, AspNetCore.Mvc and AspNetCore.StaticFiles are the packages that I added using Manage NuGet Packages dialog. Did they get added to the top just because the references from the template were already out of order?

We call CPS to Add packages, they determine where to write. I’d argue that they should write it at the end of the list, instead of the beginning.
I think that is how legacy csproj handles insertions.

```
    Bala – can you please file a bug on Sri’s team – my guess is that he’d want this bug on Roslyn-Project-System repro.

    I am going to close our bug, once I find the tracking issue for #1.
```
